sms_validator/models.py

```diff
@@ -117,21 +123,26 @@ class SMSToken(models.Model):
 
     def save(self, *args, **kwargs):
         if not self.key:
-            key = self.generate_key()
-            i = 0
-            while SMSToken.objects.filter(key=key).exists():
-                key = self.generate_key()
-                i += 1
-                if i > 1000:
-                    raise RuntimeError('Max iterations to generate key exceeded')
-            self.key = key
+            self.generate_key()
         return super(SMSToken, self).save(*args, **kwargs)
 
-    def generate_key(self):
+    def generate_key(self, key_generator=digit_token_generator):
```

Comment:
Nemělo by to být takto?

`def generate_key(self, key_generator=None):`
`    key_generator = key_generator or digit_token_generator`

Reply:
Why? I think your code is equivalent to mine, but longer.